### ISSUE TITLE
Probably Issue due to Colon rule

### ISSUE THREAD

**Issue by gagoyal** [realm/SwiftLint#3361]
### New Issue Checklist

- [ YES] Updated SwiftLint to the latest version
- [ YES] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Tried to run autocorrect on file containing below code
Example 1
```swift
guard let versionUTF8 = NSString(bytes: &systemInfo.machine, length: Int(_SYS_NAMELEN), encoding: String.Encoding.ascii.rawValue)?.utf8String,
            let versionCode = String(validatingUTF8: versionUTF8) else {
            return ""
        }

```
Example 2
```swift
        return components(separatedBy: CharacterSet.punctuationCharacters).joined()
```

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint autocorrect
```

It gets autocorrect to below version and it is incorrect!
Example 1
```swift
guard let versionUTF8 = NSString(bytes: &: stemInfo.machine, length: I: (_SYS_NAMELEN), encoding: S: ing.Encoding.ascii.rawValue)?.utf8String,
            let versionCode = String(validatingUTF8: v: sionUTF8) else {
            return ""
        }
```

Example 2
```swift
           return components(separatedBy: C: racterSet.punctuationCharacters).joined()
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? **0.40.2**
* Installation method used (Homebrew, CocoaPods, building from source, etc)? **Homebrew**
* Paste your configuration file:

```yml
disabled_rules: # rule identifiers to exclude from running
  - identifier_name
  - nesting
  - function_parameter_count
opt_in_rules: # some rules are only opt-in
  - control_statement
  - empty_count
  - trailing_newline
  - colon
  - comma

# configurable rules can be customized from this configuration file
# binary rules can set their severity level
force_cast: warning # implicitly. Give warning only for force casting

force_try:
  severity: warning # explicitly. Give warning only for force try

type_body_length:
  - 300 # warning
  - 400 # error

# or they can set both explicitly
file_length:
  warning: 500
  error: 800

large_tuple: # warn user when using 3 values in tuple, give error if there are 4
   - 3
   - 4
   
# naming rules can set warnings/errors for min_length and max_length
# additionally they can set excluded names
type_name:
  min_length: 4 # only warning
  max_length: # warning and error
    warning: 30
    error: 35
  excluded: iPhone # excluded via string
reporter: "xcode"

```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? **NO**
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? **Xcode 11.7, Build version 11E801a**
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules` **Included above**
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

**Comment by gagoyal** [comment]
Can anyone validate the issue and provide a workaround? I don't want to disable rules on multiple files or functions.
